soc/silabs/silabs_siwx917/siwx917_isp_prepare.py

```diff
@@ -1,47 +1,88 @@
+    0x8e80243d, 0xaef3d22d, 0xce67c81d, 0xee143e0d, 0x0f4ffc7d, 0x2f3c0a6d, 0x4fa8105d, 0x6fdbe64d,
+    0xc3b5618a, 0xe3c6979a, 0x83528daa, 0xa3217bba, 0x427ab9ca, 0x62094fda, 0x029d55ea, 0x22eea3fa,
+    0x14eaaf53, 0x34995943, 0x540d4373, 0x747eb563, 0x95257713, 0xb5568103, 0xd5c29b33, 0xf5b16d23,
+    0x59dfeae4, 0x79ac1cf4, 0x193806c4, 0x394bf0d4, 0xd81032a4, 0xf863c4b4, 0x98f7de84, 0xb8842894,
+    0xa7557a9b, 0x87268c8b, 0xe7b296bb, 0xc7c160ab, 0x269aa2db, 0x06e954cb, 0x667d4efb, 0x460eb8eb,
+    0xea603f2c, 0xca13c93c, 0xaa87d30c, 0x8af4251c, 0x6bafe76c, 0x4bdc117c, 0x2b480b4c, 0x0b3bfd5c,
+    0x3d3ff1f5, 0x1d4c07e5, 0x7dd81dd5, 0x5dabebc5, 0xbcf029b5, 0x9c83dfa5, 0xfc17c595, 0xdc643385,
+    0x700ab442, 0x50794252, 0x30ed5862, 0x109eae72, 0xf1c56c02, 0xd1b69a12, 0xb1228022, 0x91517632,
+    0xdd2a9970, 0xfd596f60, 0x9dcd7550, 0xbdbe8340, 0x5ce54130, 0x7c96b720, 0x1c02ad10, 0x3c715b00,
+    0x901fdcc7, 0xb06c2ad7, 0xd0f830e7, 0xf08bc6f7, 0x11d00487, 0x31a3f297, 0x5137e8a7, 0x71441eb7,
+    0x4740121e, 0x6733e40e, 0x07a7fe3e, 0x27d4082e, 0xc68fca5e, 0xe6fc3c4e, 0x8668267e, 0xa61bd06e,
+    0x0a7557a9, 0x2a06a1b9, 0x4a92bb89, 0x6ae14d99, 0x8bba8fe9, 0xabc979f9, 0xcb5d63c9, 0xeb2e95d9,
+    0x53aabd4d, 0x73d94b5d, 0x134d516d, 0x333ea77d, 0xd265650d, 0xf216931d, 0x9282892d, 0xb2f17f3d,
+    0x1e9ff8fa, 0x3eec0eea, 0x5e7814da, 0x7e0be2ca, 0x9f5020ba, 0xbf23d6aa, 0xdfb7cc9a, 0xffc43a8a,
+    0xc9c03623, 0xe9b3c033, 0x8927da03, 0xa9542c13, 0x480fee63, 0x687c1873, 0x08e80243, 0x289bf453,
+    0x84f57394, 0xa4868584, 0xc4129fb4, 0xe46169a4, 0x053aabd4, 0x25495dc4, 0x45dd47f4, 0x65aeb1e4,
+    0x29d55ea6, 0x09a6a8b6, 0x6932b286, 0x49414496, 0xa81a86e6, 0x886970f6, 0xe8fd6ac6, 0xc88e9cd6,
+    0x64e01b11, 0x4493ed01, 0x2407f731, 0x04740121, 0xe52fc351, 0xc55c3541, 0xa5c82f71, 0x85bbd961,
+    0xb3bfd5c8, 0x93cc23d8, 0xf35839e8, 0xd32bcff8, 0x32700d88, 0x1203fb98, 0x7297e1a8, 0x52e417b8,
+    0xfe8a907f, 0xdef9666f, 0xbe6d7c5f, 0x9e1e8a4f, 0x7f45483f, 0x5f36be2f, 0x3fa2a41f, 0x1fd1520f,
+]
+
+
+def calc_crc32(data: bytes) -> int:
+    register = 0
+    for b in data:
+        register = crc_table[(b ^ register) & 0xFF] ^ (register >> 8)
+    return register
 
 
 def calc_checksum(data: Union[bytes, bytearray], size: int, prev_sum: int) -> int:
     # Truncate
     data = data[:size]
     # Zero-pad data to mul of 4 bytes
     nzeros = ((len(data) + 3) // 4 * 4) - len(data)
-    data += b'\0' * nzeros
+    data += b"\0" * nzeros
     # Reinterpret data as LE u32
-    ints = list(x[0] for x in struct.iter_unpack('<I', data))
+    ints = list(x[0] for x in struct.iter_unpack("<I", data))
     # Sum
     chk = prev_sum + sum(ints)
-    # Convert to u32 and account each overflow as 1's complement addition
-    chk = (chk & 0xFFffFFff) + (chk >> 32)
-    chk = (~chk) & 0xFFffFFff
+    # Convert to u32 and account each overflow as 1"s complement addition
+    chk = (chk & 0xFFFFFFFF) + (chk >> 32)
+    chk = (~chk) & 0xFFFFFFFF
```

Comment:
Why uppercase hex instead of lowercase? `crc_table` uses lowercase. I don't have a strong preference either way (well, slightly biased towards lowercase), but seems like it would be good to at least be consistent throughout the python script.

Reply:
I share your opinion, but it seems we are not representative developers: 
https://github.com/SiliconLabsSoftware/zephyr-silabs/actions/runs/12674868242/job/35324557995?pr=75#step:6:34